Lib/test/test_import/__init__.py

```diff
@@ -2434,6 +2449,7 @@ def test_with_reinit_reloaded(self):
     # Also, we test with a single-phase module that has global state,
     # which is shared by all interpreters.
 
+    @no_rerun(reason="rerun not possible; module state is never cleared (see gh-102251)")
```

Comment:
This test is not leaking.
```suggestion
```

Reply:
It is; please see the previous buildbot run.

Reply:
See also ref leak run on Ubuntu container:

<details>

```console
root@9ad8da62d4a6:/src# git checkout e5baa024265d9489da7e354e25021f82c34ade3b
Note: switching to 'e5baa024265d9489da7e354e25021f82c34ade3b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at e5baa02426 Doh
root@9ad8da62d4a6:/src# git show HEAD
commit e5baa024265d9489da7e354e25021f82c34ade3b (HEAD)
Author: Erlend E. Aasland <erlend.aasland@protonmail.com>
Date:   Fri Jun 23 13:04:24 2023 +0200

    Doh

diff --git a/Lib/test/test_import/__init__.py b/Lib/test/test_import/__init__.py
index 938e7e636a..1114ae2218 100644
--- a/Lib/test/test_import/__init__.py
+++ b/Lib/test/test_import/__init__.py
@@ -2449,7 +2449,7 @@ def test_with_reinit_reloaded(self):
     # Also, we test with a single-phase module that has global state,
     # which is shared by all interpreters.
 
-    #@no_rerun(reason="rerun not possible; module state is never cleared (see gh-102251)")
+    @no_rerun(reason="rerun not possible; module state is never cleared (see gh-102251)")
     @requires_subinterpreters
     def test_basic_multiple_interpreters_main_no_reset(self):
         # without resetting; already loaded in main interpreter
root@9ad8da62d4a6:/src# pushd
/build /src
root@9ad8da62d4a6:/build# ./python -m test -R : test_import
0:00:00 load avg: 0.37 Run tests sequentially
0:00:00 load avg: 0.37 [1/1] test_import
beginning 9 repetitions
123456789
.........
test_import leaked [10, 10, 10, 10] references, sum=40
test_import leaked [8, 8, 8, 8] memory blocks, sum=32
test_import failed (reference leak)

== Tests result: FAILURE ==

1 test failed:
    test_import

Total duration: 29.1 sec
Tests result: FAILURE
root@9ad8da62d4a6:/build# pushd
/src /build
root@9ad8da62d4a6:/src# git checkout fix-test-import-leaks-alt
Previous HEAD position was e5baa02426 Doh
Switched to branch 'fix-test-import-leaks-alt'
Your branch is up to date with 'origin/fix-test-import-leaks-alt'.
root@9ad8da62d4a6:/src# git show HEAD
commit 0784bc19b8ba5b0f07fd6c15f35e77e10fd4d20f (HEAD -> fix-test-import-leaks-alt, origin/fix-test-import-leaks-alt)
Author: Erlend E. Aasland <erlend.aasland@protonmail.com>
Date:   Fri Jun 23 14:17:10 2023 +0200

    Also skip reruns for test_basic_multiple_interpreters_deleted_no_reset()

diff --git a/Lib/test/test_import/__init__.py b/Lib/test/test_import/__init__.py
index 1114ae2218..0a53b4b182 100644
--- a/Lib/test/test_import/__init__.py
+++ b/Lib/test/test_import/__init__.py
@@ -2518,6 +2518,7 @@ def test_basic_multiple_interpreters_main_no_reset(self):
         #  * m_copy was copied from interp2 (was from interp1)
         #  * module's global state was updated, not reset
 
+    @no_rerun(reason="rerun not possible; module state is never cleared (see gh-102251)")
     @requires_subinterpreters
     def test_basic_multiple_interpreters_deleted_no_reset(self):
         # without resetting; already loaded in a deleted interpreter
root@9ad8da62d4a6:/src# pushd
/build /src
root@9ad8da62d4a6:/build# ./python -m test -R : test_import
0:00:00 load avg: 0.32 Run tests sequentially
0:00:00 load avg: 0.32 [1/1] test_import
beginning 9 repetitions
123456789
.........

== Tests result: SUCCESS ==

1 test OK.

Total duration: 27.5 sec
Tests result: SUCCESS
root@9ad8da62d4a6:/build# 
```

</details>

Reply:
Isn't the Ubuntu ref leak run indicating `test_basic_multiple_interpreters_main_no_reset` leaking?

Reply:
Ah, yeah, of course; thanks for the heads up. I'll remove the decorator from `test_basic_multiple_interpreters_main_no_reset` before landing. Tested out locally; all ok.